drivers/wifi/infineon/cyw43xxx_drv.c

```diff
@@ -0,0 +1,698 @@
+		p_zy_result->mac_length = 6;
+		memcpy(p_zy_result->mac, &p_whd_result->BSSID, 6);
+	}
+}
+
+static void scan_callback(whd_scan_result_t **result_ptr, void *user_data, whd_scan_status_t status)
+{
+	struct cyw43xxx_wifi_runtime *data = (struct cyw43xxx_wifi_runtime *)user_data;
+	whd_scan_result_t whd_scan_result;
+	struct wifi_scan_result zephyr_scan_result;
+
+	if (status == WHD_SCAN_COMPLETED_SUCCESSFULLY || status == WHD_SCAN_ABORTED) {
+		data->scan_rslt_cb(data->iface, 0, NULL);
+
+		return;
+	}
+	if ((result_ptr != NULL) || (*result_ptr != NULL)) {
+		memcpy(&whd_scan_result, *result_ptr, sizeof(whd_scan_result_t));
+		parse_scan_result(&whd_scan_result, &zephyr_scan_result);
+		data->scan_rslt_cb(data->iface, 0, &zephyr_scan_result);
+	}
+	memset(*result_ptr, 0, sizeof(whd_scan_result_t));
+}
+
+static int cyw43xxx_mgmt_scan(const struct device *dev, struct wifi_scan_params *params,
+			      scan_result_cb_t cb)
+{
+	int ret = 0;
+	struct cyw43xxx_wifi_runtime *data = (struct cyw43xxx_wifi_runtime *)dev->data;
+
+	data->scan_rslt_cb = cb;
+
+	if (k_sem_take(&data->sema_common, K_MSEC(CYW43XXX_WAIT_SEMA_MS)) != 0) {
+		return -EAGAIN;
+	}
+
+	ret = whd_wifi_scan(cyw43xxx_if, params->scan_type, WHD_BSS_TYPE_ANY, &(data->ssid), NULL,
+			    NULL, NULL, scan_callback, &(data->scan_result), data);
+
+	k_sem_give(&data->sema_common);
+	return ret;
+}
+
+static int cyw43xxx_mgmt_connect(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct cyw43xxx_wifi_runtime *data = (struct cyw43xxx_wifi_runtime *)dev->data;
+	whd_ssid_t ssid = {0};
+
+	if (k_sem_take(&data->sema_common, K_MSEC(CYW43XXX_WAIT_SEMA_MS)) != 0) {
+		return -EAGAIN;
+	}
+
+	if (data->is_sta_connected) {
+		LOG_ERR("Already connected");
+		k_sem_give(&data->sema_common);
+		return -EALREADY;
+	}
+
+	ssid.length = params->ssid_length;
+	memcpy(ssid.value, params->ssid, params->ssid_length);
+
+	whd_scan_result_t scan_result;
+	whd_scan_result_t usr_result = {0};
+
+	usr_result.SSID.length = ssid.length;
+	memcpy(usr_result.SSID.value, ssid.value, ssid.length);
+
+	if (whd_wifi_scan(cyw43xxx_if, WHD_SCAN_TYPE_ACTIVE, WHD_BSS_TYPE_ANY, NULL, NULL, NULL,
+			  NULL, scan_cb_search, &scan_result, &(usr_result)) != WHD_SUCCESS) {
+		LOG_ERR("Failed start scan");
+		k_sem_give(&data->sema_common);
+		return -EAGAIN;
+	}
+
+	if (k_sem_take(&cyw43xxx_wifi_data.sema_scan, K_MSEC(12 * 1000)) != 0) {
+		whd_wifi_stop_scan(cyw43xxx_if);
+		k_sem_give(&data->sema_common);
+		return -EAGAIN;
+	}
+
+	if (usr_result.security == 0) {
+		k_sem_give(&data->sema_common);
+		return -EAGAIN;
+	}
+
+	/* Connect to the network */
+	if (whd_wifi_join(cyw43xxx_if, &usr_result.SSID, usr_result.security, params->psk,
+			  params->psk_length) != WHD_SUCCESS) {
+		LOG_ERR("Failed to connect with network");
+		k_sem_give(&data->sema_common);
+		return -EAGAIN;
+	}
+	data->is_sta_connected = 1;
+
+	net_eth_carrier_on(cyw43xxx_wifi_iface);
```

Comment:
Similar to the changes in #59421, carrier on should be set when the interface is initialized, with the dormant interface (`net_if_dormant_on` and `net_if_dormant_off`) used to set network association status.

Reply:
updated.

Please confirm that from shell we need call 'net iface up 1' before net operation (e.g. ping):
```
uart:~$ wifi connect SSID 4 PSWD
Connected
Connection requested

uart:~$ net iface up 1
[00:01:19.301,000] <inf> net_dhcpv4: Received: 172.20.10.2

uart:~$ net ping 172.20.10.1
PING 172.20.10.1
28 bytes from 172.20.10.1 to 172.20.10.2: icmp_seq=1 ttl=64 time=21 ms
28 bytes from 172.20.10.1 to 172.20.10.2: icmp_seq=2 ttl=64 time=12 ms
28 bytes from 172.20.10.1 to 172.20.10.2: icmp_seq=3 ttl=64 time=13 ms
```